scripts/generate.js

```diff
@@ -0,0 +1,191 @@
+    process.exit(0);
+  }
+
+  const name = args['--name'] || 'element';
+  const dir = path.resolve(
+    args['--dir'] || `./OrionStatelessComponents__ods-${lowerKebabCase(name)}`
+  );
+
+  return {
+    name,
+    dir,
+    verbose: args['--verbose'],
+  };
+};
+
+const makeFolder = async dir => {
+  if (dir !== '.' && !fs.existsSync(dir)) {
+    await fsPromises.mkdir(dir);
+  }
+};
+
+const formatTemplateFileContents = (content, { name }) => {
+  // name to lower-kebab-case (e.g. Text Input -> text-input)
+  const lowerKebabCaseName = lowerKebabCase(name);
+  // name to UpperCamelCase (e.g. text-input -> TextInput)
+  const upperCamelCaseName = upperCamelCase(name);
+  const replacements = [
+    { regex: /\[name\]/g, value: lowerKebabCaseName },
+    {
+      regex: /\[Name\]/g,
+      value: upperCamelCaseName,
+    },
+  ];
+
+  // replace all instances of [name] and [Name] accordingly
+  let result = content;
+  for (let i = 0; i < replacements.length; i++) {
+    const { regex, value } = replacements[i];
+    result = result.replace(regex, value);
+  }
+  return result;
+};
+
+const copyFile = async (sourcePath, targetPath, params, fileRenames = {}) => {
+  const stats = await fsPromises.stat(sourcePath);
+  if (stats.isDirectory()) {
+    await makeFolder(targetPath);
+    await copyAllFiles(sourcePath, targetPath, params, fileRenames);
+  } else if (stats.isFile()) {
+    const templateFileContents = await fsPromises.readFile(sourcePath, {
+      encoding: 'utf8',
+    });
+    const formattedTemplateFileContents = formatTemplateFileContents(
+      templateFileContents,
+      params
+    );
+    await fsPromises.writeFile(targetPath, formattedTemplateFileContents, {
+      encoding: 'utf8',
+    });
+
+    if (params.verbose) {
+      log(`${chalk.green('Copied')}: ${sourcePath} -> ${targetPath}`);
+    }
+  }
+};
+
+const copyAllFiles = async (
+  sourcePath,
+  targetPath,
+  params,
+  fileRenames = {}
+) => {
+  const fileNames = await fsPromises.readdir(sourcePath);
+  const fileCopyPromises = [];
+  fileNames.forEach(fileName => {
+    log(
+      `${chalk.bold('Copying')}: ${fileName} -> ${fileRenames[fileName] ||
+        fileName}`
+    );
+    fileCopyPromises.push(
+      copyFile(
+        `${sourcePath}/${fileName}`,
+        `${targetPath}/${fileRenames[fileName] || fileName}`,
+        params,
+        fileRenames
+      )
+    );
+  });
+  await Promise.all(fileCopyPromises);
+};
+
+const loadingLoop = condition => {
+  if (condition()) return;
+  process.stdout.write('.');
+  setTimeout(() => {
+    loadingLoop(condition);
+  }, 1000);
+};
+
+const generateFromTemplate = async () => {
+  const params = parseArgs();
+  await makeFolder(params.dir);
+  await copyAllFiles(paths.self.template, params.dir, params, {
+    'ods-[name]_test.html': `ods-${lowerKebabCase(params.name)}_test.html`,
+    'ods-[name].js': `ods-${lowerKebabCase(params.name)}.js`,
+    'README.temp': 'README.md',
```

Comment:
I don't think that we need this or line 160. At this point the user will have customized the [name] of the component and the proper files copied over to the new working directory. 

README.temp and .travis.temp should simply be renamed README.md and .travis.yml within the template directory and simply copied over to the new working dit.

Reply:
I figured the reason for these filenames in the Boilerplate was so that `.travis.temp` didn't fire an unintended build. Is this not an issue when not at the root of the repo?

Reply:
Travis will only pick up a .travis.yml file in the ROOT of the project. Nested ones are not seen. 

In the boilerplate project, this was at the root. Thus the .temp file name.